cmds/plugins/plugins.go

```diff
@@ -91,12 +94,13 @@ func Init(pluginRegistry *pluginregistry.PluginRegistry, log *logrus.Entry) {
 	}
 
 	// user-defined function registration
-	for _, userFunction := range userFunctions {
-		for name, fn := range userFunction {
-			if err := test.RegisterFunction(name, fn); err != nil {
-				log.Fatal(err)
+	testInitOnce.Do(func() {
+		for _, userFunction := range userFunctions {
+			for name, fn := range userFunction {
+				if err := test.RegisterFunction(name, fn); err != nil {
+					log.Fatal(err)
+				}
 			}
 		}
-	}
-
+	})
```

Comment:
actually no, the other sections operate on the arguments passed to the function, this one is global. probably needs to be refactored, but for now this will do.